src/IOUService/IOUController.cs

```diff
@@ -18,23 +24,36 @@ public string GetCurrentUserName()
         [HttpGet]
         public void SendThankYouNoteTo(string recipient)
         {
-            string me = GetCurrentUserName();
-            if (!notes.ContainsKey(me))
-            {
-                notes.Add(me, new List<string>());
-            }
-            notes[me].Add(recipient);
+            Notes.Add(new Note(me, recipient));
         }
 
         [HttpGet]
-        public List<string> GetNotesSentByMe()
+        public IEnumerable<string> GetNotesSentByMe()
         {
-            return notes[GetCurrentUserName()];
+            return Notes.Where(note => note.Sender == me).Select(note => note.ToString());
         }
 
-        //public List<string> GetMyNotes()
-        //{
-        //    return something;
-        //}
+        public IEnumerable<string> GetMyNotes()
```

Comment:
Similarily, I'd probably want to know who sent the note to me.
Return a collection of notes here.